plaidml/bridge/openvino/ops/roi_align.cpp

```diff
@@ -0,0 +1,123 @@
+           "must be Constants.";
+  }
+}
+template <typename T>
+edsl::Tensor make_tensor(DType dtype, const std::vector<int64_t>& dims, const std::vector<T>& data,
+                         const std::string& name) {
+  TensorShape shape(dtype, dims);
+  Buffer buffer(shape);
+  buffer.copy_from(data.data());
+  return edsl::Constant(buffer, name);
+}
+}  // namespace
+
+namespace PlaidMLPlugin {
+
+void registerROIAlign() {
+  registerOp("roialign", [](const Context& ctx) {
+    IE_ASSERT(ctx.operands.size() == 3);
+    auto X = ctx.operands.at(0);
+    auto* layer = ngraph::as_type<ngraph::opset3::ROIAlign>(ctx.layer);
+    auto boxes = cast_constant_operand<int32_t>(1, layer);
+    auto batch_indices = cast_constant_operand<int32_t>(2, layer);
+
+    auto pooled_h = layer->get_pooled_h();
+    auto pooled_w = layer->get_pooled_w();
+    auto sampling_ratio = layer->get_sampling_ratio();
+    auto spatial_scale = layer->get_spatial_scale();
+    auto mode = layer->get_mode();
+    auto num_rois = static_cast<int32_t>(batch_indices.size());
+
+    auto input_size = X.compute_shape().sizes();
+
+    op::PoolMode pool_mode;
+    switch (mode) {
+      case ngraph::op::v3::ROIAlign::PoolingMode::AVG:
+        pool_mode = op::PoolMode::AVG;
+        break;
+      case ngraph::op::v3::ROIAlign::PoolingMode::MAX:
+        pool_mode = op::PoolMode::MAX;
+        break;
+      default:
+        std::runtime_error("Unsupported Pooling Mode");
+    }
+
+    // Crop, Resize and Pool
+    std::vector<edsl::Tensor> pooled_rois;
+    for (int i = 0; i < num_rois; i++) {
+      auto x_1 = spatial_scale * boxes[4 * i];
+      auto y_1 = spatial_scale * boxes[4 * i + 1];
+      auto x_2 = spatial_scale * boxes[4 * i + 2];
+      auto y_2 = spatial_scale * boxes[4 * i + 3];
+      auto roi_width = std::max(spatial_scale * (x_2 - x_1), 1.0f);
+      auto roi_height = std::max(spatial_scale * (y_2 - y_1), 1.0f);
+
+      int pool_size_h, pool_size_w;
+      if (sampling_ratio == 0) {
+        pool_size_h = std::ceil(roi_height / pooled_h);
+        pool_size_w = std::ceil(roi_width / pooled_w);
+      } else {
+        pool_size_h = pool_size_w = sampling_ratio;
+      }
+      auto sampling_h = pool_size_h * pooled_h;
+      auto sampling_w = pool_size_w * pooled_w;
+
+      auto interval_h = roi_width / (sampling_h + 1.0);
+      auto interval_w = roi_width / (sampling_w + 1.0);
+
+      std::vector<int32_t> indices_h(sampling_h);
+      for (int ih = 0; ih < sampling_h; ih++) {
+        indices_h[ih] = x_1 + interval_h * ih;
+      }
+      auto ind_tensor_h = make_tensor(DType::INT32, {sampling_h}, indices_h, "indices_h");
+      std::vector<int64_t> indices_w(sampling_w);
+      for (int iw = 0; iw < sampling_w; iw++) {
+        indices_w.push_back(x_1 + interval_w * iw);
+      }
+      auto ind_tensor_w = make_tensor(DType::INT32, {sampling_w}, indices_w, "indices_w");
+
+      auto slice_X = op::slice(X)
```

Comment:
Can you replace slice by gather like in 
https://github.com/Flex-plaidml-team/plaidml/pull/27/commits/a539e5a7b7d80d7741988bfa468cb4990d155279#diff-87d0ac961bc6d57d9c8fa093a5170d153165089a2234046ed414b1b5980d00b3L91
and avoid using compute_shape in
https://github.com/Flex-plaidml-team/plaidml/pull/21/files#diff-75ec0f222589cdf13f5ca902c3f223f7c0045cd1c4ed55f78a86afe59f7b9165R56